lib/web_ui/lib/src/engine/profiler.dart

```diff
@@ -7,6 +7,19 @@ part of engine;
 
 typedef OnBenchmark = void Function(String name, num value);
 
+typedef Action<R> = R Function();
```

Comment:
Did you intend to have profiler.dart change in this PR ?

Reply:
I will remove it once I rebase on top of https://github.com/flutter/engine/pull/17852. I will rebase on top of it and remove the profiler changes here. I only needed it to verify the impact of the change on preroll, where we do a ton of vector math.